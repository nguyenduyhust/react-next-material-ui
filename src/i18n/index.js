const NextI18Next = require('next-i18next').default;
const path = require('path');

module.exports = new NextI18Next({
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
});
