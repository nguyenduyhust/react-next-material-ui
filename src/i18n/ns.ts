export enum NamespaceEnum {
  COMMON = 'common',
  HOME_PAGE = 'homepage',
}
export const ns: string | Array<string> = [NamespaceEnum.COMMON, NamespaceEnum.HOME_PAGE];
export const defaultNS: string = NamespaceEnum.COMMON;
