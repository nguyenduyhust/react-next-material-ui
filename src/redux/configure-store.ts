import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import {
  typePendingReducerSet,
  createTypeReduxInitialState,
  typeReduxMiddleware,
} from './type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './reducers/app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface RootState {
  appState: AppReducer.State;
}

export const InitialState: RootState = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: RootState) => {
  return createStore(
    rootReducer,
    { ...InitialState, ...initialState },
    composeWithDevTools(applyMiddleware(...middlewares)),
  );
};
