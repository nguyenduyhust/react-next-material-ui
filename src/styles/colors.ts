export const BLACK_COLOR = '#000000';
export const WHITE_COLOR = '#FFFFFF';
export const TORY_BLUE_COLOR = '#344090';
export const NAVY_BLUE_COLOR = '#1976D2';
export const PATTENS_BLUE_COLOR = '#D8DFE5';
export const MID_NIGHT_COLOR = '#233044';
export const ALICE_BLUE_COLOR = '#F7F9FC';
export const NOBEL_COLOR = '#9E9E9E';
export const CHARCOAL_COLOR = '#424242';
