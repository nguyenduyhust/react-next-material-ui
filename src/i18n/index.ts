import NextI18Next from 'next-i18next';
import path from 'path';
import moment from 'moment';
import { ns, defaultNS } from './ns';

const {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
} = new NextI18Next({
  ns,
  defaultNS,
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
  interpolation: {
    format: function (value, format, lng) {
      if (value instanceof Date) {
        return moment(value).format(format);
      }
      return value;
    },
  },
});

export {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
};
export { NamespaceEnum } from './ns';
