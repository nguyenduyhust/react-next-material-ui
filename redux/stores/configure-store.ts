import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import { typePendingReducerSet, createTypeReduxInitialState, typeReduxMiddleware } from 'type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface Store {
  appState: AppReducer.State;
}

export const InitialState: Store = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: Store) => {
  return createStore(rootReducer, { ...InitialState, ...initialState }, composeWithDevTools(applyMiddleware(...middlewares)));
};
