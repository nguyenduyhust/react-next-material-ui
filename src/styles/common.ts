import { CSSProperties } from '@material-ui/styles';

export const link: CSSProperties = {
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
};
