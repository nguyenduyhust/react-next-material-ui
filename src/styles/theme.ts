import { createMuiTheme, Theme } from '@material-ui/core/styles';
import {
  BLACK_COLOR,
  WHITE_COLOR,
  TORY_BLUE_COLOR,
  NAVY_BLUE_COLOR,
  PATTENS_BLUE_COLOR,
  MID_NIGHT_COLOR,
  ALICE_BLUE_COLOR,
  NOBEL_COLOR,
  CHARCOAL_COLOR,
} from '../constants/colors';
import {
  XS_BREAKPOINT,
  SM_BREAKPOINT,
  MD_BREAKPOINT,
  LG_BREAKPOINT,
  XL_BREAKPOINT,
} from '../constants/breakpoints';

const theme: Theme = createMuiTheme({
  colors: {
    black: BLACK_COLOR,
    white: WHITE_COLOR,
    toryBlue: TORY_BLUE_COLOR,
    navyBlue: NAVY_BLUE_COLOR,
    pattensBlue: PATTENS_BLUE_COLOR,
    midnight: MID_NIGHT_COLOR,
    aliceBlue: ALICE_BLUE_COLOR,
    nobel: NOBEL_COLOR,
    charcoal: CHARCOAL_COLOR,
  },
  breakpoints: {
    values: {
      xs: XS_BREAKPOINT,
      sm: SM_BREAKPOINT,
      md: MD_BREAKPOINT,
      lg: LG_BREAKPOINT,
      xl: XL_BREAKPOINT,
    },
  },
  spacing: 8,
  palette: {
    primary: {
      main: NAVY_BLUE_COLOR,
    },
  },
});

export default theme;
