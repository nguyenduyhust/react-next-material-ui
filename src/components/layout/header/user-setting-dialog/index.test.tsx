import React from 'react';
import { mount } from 'enzyme';
import firebase from 'firebase';
import { I18nextProvider } from 'react-i18next';
import { SnackbarProvider } from 'notistack';
import { Provider } from 'react-redux';
import { configureStore } from '~/redux/configure-store';
import { i18n } from '~/i18n';
import { ThemeProvider } from '@material-ui/core/styles';
import theme from '~/styles/theme';
import Component from '.';

it('renders correctly', () => {
  const onClose = jest.fn();
  const user = {
    uid: 'uid',
    email: 'test@test.com',
    displayName: 'Test',
    photoURL: 'http://test.com',
  } as firebase.User;
  const store = configureStore();

  const tree = mount(
    <SnackbarProvider>
      <I18nextProvider i18n={i18n}>
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <Component onClose={onClose} user={user} open={true} />
          </Provider>
        </ThemeProvider>
      </I18nextProvider>
    </SnackbarProvider>,
  ).html();
  expect(tree).toMatchSnapshot();
});
