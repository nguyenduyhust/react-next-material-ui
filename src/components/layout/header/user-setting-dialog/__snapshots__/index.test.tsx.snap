// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollBody\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper makeStyles-dialogPaper-2 MuiDialog-paperScrollBody MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogTitle-root makeStyles-dialogTitle-3\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">user_setting</h2></div><form><div class=\\"MuiDialogContent-root makeStyles-dialogContent-4\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"makeStyles-avatar-6\\"><img src=\\"http://test.com\\" alt=\\"\\"><input type=\\"file\\" hidden=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-editAvatarButton-7 MuiIconButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required\\" data-shrink=\\"true\\" for=\\"display-name\\" id=\\"display-name-label\\">display_name<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"display-name\\" id=\\"display-name\\" name=\\"displayName\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"Test\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13\\"><span>display_name&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled Mui-required Mui-required\\" data-shrink=\\"true\\">email_address<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" disabled=\\"\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled\\" value=\\"test@test.com\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13\\"><span>email_address&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required\\" data-shrink=\\"true\\">language<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\" aria-labelledby=\\"mui-component-select-language\\" id=\\"mui-component-select-language\\">english</div><input name=\\"language\\" aria-hidden=\\"true\\" tabindex=\\"-1\\" class=\\"MuiSelect-nativeInput\\" required=\\"\\" value=\\"en\\"><svg class=\\"MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13\\"><span>language&nbsp;*</span></legend></fieldset></div></div></div></div></div><div class=\\"MuiDialogActions-root makeStyles-dialogActions-5 MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">cancel</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitButton-8 MuiButton-containedPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">submit</span></button></div></form></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
